src/ServicePulse.Host/app/js/views/message/controller.js

```diff
@@ -49,25 +50,41 @@
             if (!message)
                 return false;
 
-            if (!angular.isDefined(message.messageBody)) {
-                serviceControlService.getMessageBody(message.message_id).then(function (msg) {
-                    message.messageBody = msg.data;
-                }, function () {
-                    message.bodyUnavailable = "message body unavailable";
-                });
-            }
-
             if (!angular.isDefined(message.messageHeaders)) {
                 serviceControlService.getMessageHeaders(message.message_id).then(function (msg) {
                     message.messageHeaders = msg.data[0].headers;
                 }, function () {
                     message.headersUnavailable = "message headers unavailable";
                 });
             }
+
+            if (angular.isDefined(message.messageHeaders) && !angular.isDefined(message.messageBody)) {
```

Comment:
The reason why I added: `angular.isDefined(message.messageHeaders)`
Is that I need headers to display body properly. I could either put whole this section to handler of loading headers, Or push the rendering concept of the message body a bit further. At the moment I decided to do the if's in controller not in the view to make it a bit more clear. I am open to suggestions as I don't like this piece.

The whole method is called on every change of panel.